### PR TITLE
Refresher sets proper OrchestrationTemplate type

### PR DESCRIPTION
**Pull request by bzwei**
This is to complete the work from #76. The refresher will need to set type to the new Openstack Orchestration Template type.

Previously depending on the format (json/yaml) of the template refresher sets the type to be either `OrchestrationTemplateCfn` or `OrchestrationTemplateHot`. Now they are unified to `ManageIQ::Providers::Openstack::CloudManager::OrchestrationTemplate`. Therefore the class itself is updated to support both json and yaml formats.

Note currently we don't have  `ManageIQ::Providers::Openstack::InfraManager::OrchestrationTemplate` class. So the template collected from an infra stack will be `CloudManager::OrchestrationTemplate`. Is this acceptable?

### CODE REVIEW

**Comment by bzwei** [comment]
@miq-bot add_label refactoring
@miq-bot add_label enhancement
@miq-bot assign @tzumainn 
cc @Ladas

File: app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb [hunk 1]
```diff
@@ -220,7 +220,7 @@ def orchestration_template(stack)
     template = collector.orchestration_template(stack)
     if template
       o = persister.orchestration_templates.find_or_build(stack.id)
-      o.type = stack.template.format == "HOT" ? "OrchestrationTemplateHot" : "OrchestrationTemplateCfn"
```
Comment (Ladas) [inline app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb]: @bzwei why did we have CFN template format here, when people are not using HOT?

Comment (bzwei) [reply app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb]: Yes, openstack can take CFN template too. With this refactored class, now Openstack's template support both JSON (CFN) and YAML (HOT) format.

**Comment by Ladas** [comment]
@bzwei hm, it should be ok to use the safe_load, right? Since this is user input, we should porbably load it safely.

**Comment by bzwei** [comment]
@Ladas A typical template contains string like this `version: 2013-05-13`
`safe_load` causes an error: `*** Psych::DisallowedClass Exception: Tried to load unspecified class: Date`
`load` has no error but parses the string as a date object, which is not of our interest but at least does not fail us.

**Comment by Ladas** [comment]
@bzwei right, I think we will have to solve it a different way. The load should not be used for user input data, it allows remote code execution (check the example in) https://brakemanscanner.org/docs/warning_types/remote_code_execution_yaml_load

File: app/models/manageiq/providers/openstack/cloud_manager/orchestration_template.rb [hunk 2]
```diff
@@ -65,14 +65,37 @@ def self.eligible_manager_types
...
+  # returns either json or yaml
+  def format
+    content.strip.start_with?('{') ? 'json'.freeze : 'yaml'.freeze
+  end
+
+  private
+
+  def parse
+    return JSON.parse(content) if format == 'json'
+    YAML.load(content)
```
Comment (Ladas) [inline app/models/manageiq/providers/openstack/cloud_manager/orchestration_template.rb]: @bzwei you should be able to use  YAML.safe_load(content, [Date])

**Comment by bzwei** [comment]
@Ladas fixed

**Comment by Ladas** [comment]
@tzumainn looks good to me now, can you guys review?

**Comment by Ladas** [comment]
ping @tzumainn @aufi

**Comment by aufi** [comment]
Looks good to me, merging.

**Pull request merged**